### PR TITLE
Fix #1779: Updated placeholder schema

### LINKED ISSUE #1779: Placeholder is not getting parsed in Input Interactions
Currently, if we check the app on alpha content, the placeholder in `TextInput` interaction does not show any placeholder value, instead it is empty as shown below. 
Make sure that all input interactions parse placeholder text correctly.

Screenshot_1599097704

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Fixes #1779 

This PR updates the schema of `placeholder` text.

NOTE: The placeholder-texts might be slightly different but that should not be an issue because this is dummy content.

Also, I think we should add one test in `app` module related to hint visibility but that  depends on #1630 getting merged.

Result can be seen in the form of hints in various explorations.

Screenshot_1599102839

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt [hunk 1]
```diff
@@ -66,6 +66,6 @@ class TextInputViewModel(
 
   private fun deriveHintText(interaction: Interaction): CharSequence {
     // The default placeholder for text input is empty.
-    return interaction.customizationArgsMap["placeholder"]?.normalizedString ?: ""
+    return interaction.customizationArgsMap["placeholder"]?.subtitledUnicode?.unicodeStr.toString()
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt]: Will we always be using subtitled unicode or do we need to support multiple types? Can you confirm with the backend definitions for the interactions?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @rt4914! Do NumericInput and FractionInput also need to change?

File: app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt]: `NumericInput` does not use `customizationArgs`. Reference: https://github.com/oppia/oppia/blob/acd87b4995ebe455ac0438603e2d51649b26756f/extensions/interactions/NumericInput/NumericInput.py#L41

`FractionInput` uses entirely different schema which has already been parsed correctly in `StateRetriever` class.
Reference: https://github.com/oppia/oppia/blob/acd87b4995ebe455ac0438603e2d51649b26756f/extensions/interactions/FractionInput/FractionInput.py#L38

`TextInput` uses only `SubtitledUnicode` for placeholder. Reference: https://github.com/oppia/oppia/blob/acd87b4995ebe455ac0438603e2d51649b26756f/extensions/interactions/TextInput/TextInput.py#L52

Comment (rt4914) [reply app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt]: Please do let me know if I am checking the incorrect code in web.

Comment (BenHenning) [reply app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt]: That looks correct, and I think confirms that more work might be needed. Fractions, for example, defines its placeholder type as subtitledHtml (https://github.com/oppia/oppia/blob/acd87b4995ebe455ac0438603e2d51649b26756f/extensions/interactions/FractionInput/FractionInput.py#L64) but we're assuming it's normalized string (https://github.com/oppia/oppia-android/blob/develop/app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt#L102). This seems to require an update.

Comment (rt4914) [reply app/src/main/java/org/oppia/app/player/state/itemviewmodel/TextInputViewModel.kt]: Updated. Check last 2 commits. One is related to schema update and other one for test-TODO.

File: app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt [hunk 2]
```diff
@@ -1444,7 +1445,8 @@ class StateFragmentTest {
 
     fun inject(stateFragmentTest: StateFragmentTest)
 
-    @IsOnRobolectric fun isOnRobolectric(): Boolean
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt:1447]: FYI this is technically conformant to our style guide: https://developer.android.com/kotlin/style-guide#annotations & ktlint normally seems amenable to it.

**Review by BenHenning — APPROVED** [review]
Thanks @rt4914! LGTM--my only comment isn't suggesting any changes, so feel free to resolve after reading it.

**Comment by BenHenning** [comment]
Actually @rt4914 going to go ahead and merge this since it'd be nice to pick it up in the release.

**Pull request merged**